src/components/Sidebar/Sidebar.tsx

```diff
@@ -8,6 +8,51 @@ import SidebarItem from './SidebarItem';
 export default function Sidebar({ checkboxId }: { checkboxId: string }) {
   const { user, signOut, signIn, isAuthenticated, loading } = useAuth();
 
+  // Lock the UI into a 'loading' state during sign-in/out transitions to prevent component flickering/flashes.
+  const [authAction, setAuthAction] = useState<null | 'signin' | 'signout'>(
```

Comment:
Typically, you'd want to put a boolean value for 'loading' into the state instead of what you have here.